plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,182 @@
+	if err != nil {
+		return fmt.Errorf("failed to get processes info: %v", err)
+	}
+
+	// API call to get identification string
+	err = s.rpcClient.Call("supervisor.getIdentification", nil, &status.Ident)
+	if err != nil {
+		return fmt.Errorf("failed to get instance identification: %v", err)
+	}
+
+	// Iterating through array of structs with processes info and adding fields to accumulator
+	for _, process := range rawProcessData {
+		processTags, processFields, err := s.parseProcessData(process, status)
+		if err != nil {
+			acc.AddError(err)
+			continue
+		}
+		acc.AddFields("supervisor_processes", processFields, processTags)
+	}
+	// Adding instance info fields to accumulator
+	instanceTags, instanceFields, err := s.parseInstanceData(status)
+	if err != nil {
+		return fmt.Errorf("failed to parse instance data: %v", err)
+	}
+	acc.AddFields("supervisor_instance", instanceFields, instanceTags)
+	return nil
+}
+
+func (s *Supervisor) parseProcessData(pInfo processInfo, status supervisorInfo) (map[string]string, map[string]interface{}, error) {
+	tags := map[string]string{
+		"process": pInfo.Name,
+		"group":   pInfo.Group,
+	}
+	fields := map[string]interface{}{
+		"uptime": pInfo.Now - pInfo.Start,
+		"state":  pInfo.State,
+	}
+	if s.fieldFilter.Match("pid") {
+		fields["pid"] = pInfo.Pid
+	}
+	if s.fieldFilter.Match("rc") {
+		fields["exitCode"] = pInfo.ExitStatus
+	}
+	if s.UseIdentTag {
+		tags["server"] = status.Ident
+	} else {
+		var err error
+		tags["server"], err = beautifyServerString(s.Server)
+		if err != nil {
+			return map[string]string{}, map[string]interface{}{}, err
+		}
+	}
+	return tags, fields, nil
+}
+
+// Parsing of supervisor instance data
+func (s *Supervisor) parseInstanceData(status supervisorInfo) (map[string]string, map[string]interface{}, error) {
+	server := status.Ident
+	// Using server URL for server tag instead of instance identification, if plugin configured accordingly
+	if !s.UseIdentTag {
+		var err error
+		server, err = beautifyServerString(s.Server)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to decorate server string: %v", err)
+		}
+	}
+	tags := map[string]string{"server": server}
+	fields := map[string]interface{}{"state": status.StateCode}
+	return tags, fields, nil
+}
+
+func (s *Supervisor) Init() error {
+	// Using default server URL if none was specified in config
+	if s.Server == "" {
+		s.Server = "http://localhost:9001/RPC2"
+	}
+	var err error
+	// Initializing XML-RPC client
+	s.rpcClient, err = xmlrpc.NewClient(s.Server, nil)
+	if err != nil {
+		return fmt.Errorf("XML-RPC client initialization failed: %v", err)
+	}
+	// Setting filter for additional metrics
+	s.fieldFilter, err = filter.NewIncludeExcludeFilter(s.MetricsInc, s.MetricsExc)
+	if err != nil {
+		return fmt.Errorf("metrics filter setup failed: %v", err)
+	}
+	return nil
+}
+
+func init() {
+	inputs.Add("supervisor", func() telegraf.Input {
+		return &Supervisor{}
```

Comment:
I think your default is `metrics_exclude = ["pid", "rc"]` so you should reflect that in the code
```suggestion
		return &Supervisor{MetricsExc:  ["pid", "rc"]}
```

Reply:
Done, thanks for your attentiveness.

Reply:
That's what I'm here for. ;-)